autonomi/src/client/data_types/chunk.rs

```diff
@@ -110,13 +104,34 @@ fn hash_to_short_string(input: &str) -> String {
 impl Client {
     /// Get a chunk from the network.
     pub async fn chunk_get(&self, addr: ChunkAddress) -> Result<Chunk, GetError> {
+        let result = self.chunk_get_inner(addr).await;
+        // Reporting
+        if let Some(channel) = self.client_event_sender.as_ref() {
+            let xor_name = addr.xorname();
+            let event = if result.is_ok() {
+                ClientEvent::DownloadSucceeded(*xor_name)
+            } else {
+                ClientEvent::DownloadFailed(*xor_name)
+            };
```

Comment:
Here the error is swallowed so we will have no idea why get failed

Reply:
Do you mean pass the error msg through the `Event` channel? I don't think `Error` implement `Copy` for us to do that :/ I'm not sure if there is a better alternative.